src/collections/maple_tree/mod.rs

```diff
@@ -104,66 +131,307 @@ impl<K, V> SparseLeaf<K, V>
+        let mut sl: SparseLeaf<usize, usize> = SparseLeaf::new();
+
+        // Insert K:V pair
+        sl.insert(2, 2);
 
         // update inplace.
         sl.update(2, 3);
 
-        // split?
-        // compact/sort
-        // verify
-        // clone
-        // eq?
-        // pass
+        // check that the value was correctly changed
+        assert!(sl.search(&2) == Some(&3));
     }
 
-}
+    #[test]
+    fn test_sparse_leaf_insert() {
+        let mut sl: SparseLeaf<usize, usize> = SparseLeaf::new();
 
+        // insert
+        sl.insert(2, 2);
+
+        // test valid search
+        assert!(sl.search(&2) == Some(&2));
 
+        // test invalid search
+        assert!(sl.search(&1).is_none());
+
+        // test insert after node is already full
+
+        for i in 1..8 {
+            sl.insert(i, i);
+        }
+
+        assert!(sl.insert(8, 8).is_none())
+    }
+
+    #[test]
+    fn test_sparse_leaf_remove() {
+        let mut sl: SparseLeaf<usize, usize> = SparseLeaf::new();
+
+        // check removing a non-existent value fails
+        assert!(sl.remove(&0).is_none());
+
+        // check removing a value that exists
+        sl.insert(0, 0);
+        assert!(sl.remove(&0).is_some());
+    }
+
+    #[test]
+    fn test_sparse_leaf_check_min_max() {
+        
+        let mut sl: SparseLeaf<usize, usize> = SparseLeaf::new();
+
+        // test that check_min_max returns none when the sparseLeaf is empty
+        assert!(sl.check_min_max(&0, &8).is_none());
+       
+        // insert 8 values from 0 - 7 
+        for i in 0..8 {
+            sl.insert(i, i);
+        }
+        
+        assert!(sl.check_min_max(&0, &8).is_none());
+
+        for i in 0..8{
+            sl.remove(&i);
+        }
+        
+        // test that check_min_max returns some when the values are out of the range
+        // and returns the first value that is found outside the range.
+        
+        sl.insert(10, 10);
+        sl.insert(11, 11);
+        sl.insert(12, 12);
+        assert!(sl.check_min_max(&0, &8).is_some());
+
+    }
+
+    #[test]
+    fn test_sparse_leaf_sort() {
+       
+        // test sorting full node
+        let mut sl: SparseLeaf<usize, usize> = SparseLeaf::new();
+        let mut testVals: [usize; 8] = [3, 8, 7, 4, 2, 1, 5, 6];
+        let mut sortedTestVals: [usize; 8] = [1, 2, 3, 4, 5, 6, 7, 8];
+        
+        for val in testVals.iter(){
+            sl.insert(*val, *val);
+        }
+
+        sl.sort();
+        
+        for i in 0..8{
+            
+            assert!(sl.key[i] == M::Some(sortedTestVals[i]));
+        }
+
+        // test sorting half full node with M::None's inbetween
+        
+        testVals = [3, 8, 4, 6, 2, 7, 1, 5];
+ 
+        sl = SparseLeaf::new();
+        let mut i = 0;
+        for val in testVals.iter(){
+            
+
+            // in even indices insert values and keys
+            // in odd indices insert and remove values and keys so they are left as M::None
```

Comment:
Not sure I understand this comment, can you clarify what it's intent is for the test?